src/fsharp/FSharp.Core/tasks.fs

```diff
@@ -0,0 +1,424 @@
+        // A using statement is just a try/finally with the finally block disposing if non-null.
+        builder.TryFinally(
+            (fun () -> __expand_body disp),
+            (fun () -> if not (isNull (box disp)) then disp.Dispose()))
+
+    member inline builder.For(sequence : seq<'T>, __expand_body : 'T -> TaskStep<unit, 'TOverall>) : TaskStep<unit, 'TOverall> =
+        // A for loop is just a using statement on the sequence's enumerator...
+        builder.Using (sequence.GetEnumerator(), 
+            // ... and its body is a while loop that advances the enumerator and runs the body on each element.
+            (fun e -> builder.While((fun () -> e.MoveNext()), (fun () -> __expand_body e.Current))))
+
+    member inline __.ReturnFrom (task: Task<'T>) : TaskStep<'T, 'T> =
+        let CONT = __newEntryPoint ()
+        if task.IsCompleted then
+            __entryPoint CONT
+            ret (task.GetAwaiter().GetResult())
+        else
+            Helpers.Await __machineAddr<TaskStateMachineTemplate<'T>> ((task.GetAwaiter() :> ICriticalNotifyCompletion), CONT)
+            TaskStep<'T, 'T>(false)
+
+[<AutoOpen>]
+module ContextSensitiveTasks =
+
+    let task = TaskBuilder()
+
+    [<Sealed>]
+    type Witnesses() =
+
+        interface IPriority1
+        interface IPriority2
+        interface IPriority3
+
+        // Give the type arguments explicitly to make it match the signature precisely
+        static member inline CanBind< ^TaskLike, ^TResult1, 'TResult2, ^Awaiter , 'TOverall
+                                            when  ^TaskLike: (member GetAwaiter:  unit ->  ^Awaiter)
+                                            and ^Awaiter :> ICriticalNotifyCompletion
+                                            and ^Awaiter: (member get_IsCompleted:  unit -> bool)
+                                            and ^Awaiter: (member GetResult:  unit ->  ^TResult1)>(_priority: IPriority2, taskLike : ^TaskLike, __expand_continuation: (^TResult1 -> TaskStep<'TResult2, 'TOverall>)) : TaskStep<'TResult2, 'TOverall>
+                  = TaskLikeBind<'TResult2>.GenericAwait< ^TaskLike, ^Awaiter, ^TResult1, 'TOverall> (taskLike, __expand_continuation)
+
+        static member inline CanBind (_priority: IPriority1, task: Task<'TResult1>, __expand_continuation: ('TResult1 -> TaskStep<'TResult2, 'TOverall>)) : TaskStep<'TResult2, 'TOverall>
+                  = bindTask task __expand_continuation                      
+
+        static member inline CanBind (_priority: IPriority1, computation  : Async<'TResult1>, __expand_continuation: ('TResult1 -> TaskStep<'TResult2, 'TOverall>)) : TaskStep<'TResult2, 'TOverall> 
+                  = bindTask (Async.StartAsTask computation) __expand_continuation     
+
+        // Give the type arguments explicitly to make it match the signature precisely
+        static member inline CanReturnFrom< ^TaskLike, ^Awaiter, ^T
```

Comment:
It seems indeed ReturnFrom (the let bound function) isn't used for CanReturnFrom